pkg/controller/tas/resource_flavor_test.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tas
+
+import (
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestNodeSchedulingPropertiesChanged(t *testing.T) {
+	now := metav1.Now()
+	later := metav1.NewTime(now.Add(10 * time.Second))
+
+	baseNode := &corev1.Node{
```

Comment:
We should leverage NodeWrapper: https://github.com/kubernetes-sigs/kueue/blob/5106fb28c399ad05de55ca313db0238caa060c3a/pkg/util/testingjobs/node/wrappers.go